src/utils/SessionLock.ts

```diff
@@ -171,6 +171,27 @@ export async function getSessionLock(onNewInstance: () => Promise<void>): Promis
         }
     }
 
+    // handler for pagehide and unload events, used later
+    function onPagehideEvent(): void {
+        // only remove the ping if we still think we're the owner. Otherwise we could be removing someone else's claim!
+        if (lockServicer !== null) {
+            prefixedLogger.debug("page hide: clearing our claim");
+            window.clearInterval(lockServicer);
```

Comment:
Looks like this is new.

On balance, adding it is probably a good idea.

Previously I left the lockServicer in place, which would mean if the page came back to life, we would reclaim the lock. But that is racy: if we clear our claim, but the lockServicer fires again before the tab actually dies, we'll end up with a stuck claim.



It was somewhat

Reply:
Yes, this was my motivation  - with the two handlers the risk to have a racy access might be higher. But I did not think about the page after being restored.

To close the bfache hole, couldn't we listen to `pageshow` with `event.persisted==true` and immediately re-start the lock servicer? But this might induce other problems, e.g. 
1. have one Element instance in bfcache
2. open another instance in another tab (which works after the lock is expired)
3. restore the first instance of the bfcache

Maybe listening for `pageshow` with `event.persisted==true` and somehow forcing the session to be re-initialized would be better (or just reloading the whole app?).

Reply:
Yeah, there are no ideal solutions here. To be honest, my understanding is that the presence of an `unload` handler prevents the app being eligible for the bfcache anyway, so it's probably fine in practice. Longer term, the "right" solution is to fix multi-tab loading so we don't need any of this locking business (https://github.com/vector-im/element-web/issues/26231).